### DISCUSSION TITLE
Option to choose branches while running installation scripts.

### DISCUSSION THREAD

**Discussion by charliie-dev** [ayamir/nvimdots#675] [Ideas]
Propose: as title.

Motivation: Current setting is sufficient enough for average users. However, when it comes to the time of migrating to a newer nvim version or a large to-be-tested patch-branch, an option to choose branches is quite convenient. Personally I used to do a full rundown on our installation scripts to test if there's anything weird.

prototype code:
```sh
curl -L https://api.github.com/repos/ayamir/nvimdots/branches | jq '.[].name'
```

![圖片](https://user-images.githubusercontent.com/32497323/233546654-366b77ca-0baf-4fe6-9498-20ea7f9e8790.png)

```sh
curl -L https://api.github.com/repos/ayamir/nvimdots/branches | grep name
```
btw my `grep` is an alias to `ugrep`

![圖片](https://user-images.githubusercontent.com/32497323/233549116-55dae889-6aa2-45e9-a4a9-5802971644f6.png)

Original discussion:
https://github.com/ayamir/nvimdots/issues/674#issuecomment-1517258639
https://github.com/ayamir/nvimdots/issues/674#issuecomment-1517457543

cc @ayamir @Jint-lzxy if u guys think it's worthy to add. I'll do a PR when I got time.

**Comment by Jint-lzxy** [comment]
Some thoughts (taken from https://github.com/ayamir/nvimdots/issues/674#issuecomment-1517457543):

I can't think of a reason why we should implement this - FWIW branch detection based on `nvim --version` would suffice.
In fact, if ppl really want to customize their clone branch, `CCLONE_BRANCH` is reserved as a global option for them:
https://github.com/ayamir/nvimdots/blob/f1c5f091c5f4fd8d41c92d29085f96984ba1aebb/scripts/install.ps1#L22

**Reply by charliie-dev** [reply]
> Some thoughts (taken from https://github.com/ayamir/nvimdots/issues/674#issuecomment-1517457543):
> 
> 
> 
> I can't think of a reason why we should implement this - FWIW branch detection based on `nvim --version` would suffice.
> 
> In fact, if ppl really want to customize their clone branch, `CCLONE_BRANCH` is reserved as a global option for them:
> 
> https://github.com/ayamir/nvimdots/blob/f1c5f091c5f4fd8d41c92d29085f96984ba1aebb/scripts/install.ps1#L22

i just realized what u mean lol, and yes that should work!

**Reply by charliie-dev** [reply]
Just tested it. Setting `pwsh` env var indeed works fine!
